### ISSUE TITLE
Did you retrained the sync_expert network?

### ISSUE THREAD

**Issue by yihe1003** [primepake/wav2lip_288x288#4]
hi ，did you retained the lipsync_expert , i use your training code and reload the oringal pretrained lipsync_expert.pth(down load from https://github.com/Rudrabha/Wav2Lip) , and i get an shape missmatch error as follows:

	Missing key(s) in state_dict: "face_encoder.0.act.weight", "face_encoder.1.act.weight", "face_encoder.2.act.weight", "face_encoder.3.act.weight", "face_encoder.4.act.weight", "face_encoder.5.act.weight", "face_encoder.6.act.weight", "face_encoder.7.act.weight", "face_encoder.8.act.weight", "face_encoder.9.act.weight", "face_encoder.10.act.weight", "face_encoder.11.act.weight", "face_encoder.12.act.weight", "face_encoder.13.act.weight", "face_encoder.14.act.weight", "face_encoder.15.act.weight", "face_encoder.16.act.weight", "face_encoder.17.conv_block.0.weight", "face_encoder.17.conv_block.0.bias", "face_encoder.17.conv_block.1.weight", "face_encoder.17.conv_block.1.bias", "face_encoder.17.conv_block.1.running_mean", "face_encoder.17.conv_block.1.running_var", "face_encoder.17.act.weight", "face_encoder.18.conv_block.0.weight", "face_encoder.18.conv_block.0.bias", "face_encoder.18.conv_block.1.weight", "face_encoder.18.conv_block.1.bias", "face_encoder.18.conv_block.1.running_mean", "face_encoder.18.conv_block.1.running_var", "face_encoder.18.act.weight", "face_encoder.19.conv_block.0.weight", "face_encoder.19.conv_block.0.bias", "face_encoder.19.conv_block.1.weight", "face_encoder.19.conv_block.1.bias", "face_encoder.19.conv_block.1.running_mean", "face_encoder.19.conv_block.1.running_var", "face_encoder.19.act.weight", "face_encoder.20.conv_block.0.weight", "face_encoder.20.conv_block.0.bias", "face_encoder.20.conv_block.1.weight", "face_encoder.20.conv_block.1.bias", "face_encoder.20.conv_block.1.running_mean", "face_encoder.20.conv_block.1.running_var", "face_encoder.20.act.weight", "audio_encoder.0.act.weight", "audio_encoder.1.act.weight", "audio_encoder.2.act.weight", "audio_encoder.3.act.weight", "audio_encoder.4.act.weight", "audio_encoder.5.act.weight", "audio_encoder.6.act.weight", "audio_encoder.7.act.weight", "audio_encoder.8.act.weight", "audio_encoder.9.act.weight", "audio_encoder.10.act.weight", "audio_encoder.11.act.weight", "audio_encoder.12.act.weight", "audio_encoder.13.act.weight". 
	size mismatch for face_encoder.1.conv_block.0.weight: copying a param with shape torch.Size([64, 32, 5, 5]) from checkpoint, the shape in current model is torch.Size([32, 32, 5, 5]).
	size mismatch for face_encoder.1.conv_block.0.bias: copying a param with shape torch.Size([64]) from checkpoint, the shape in current model is torch.Size([32]).
	size mismatch for face_encoder.1.conv_block.1.weight: copying a param with shape torch.Size([64]) from checkpoint, the shape in current model is torch.Size([32]).
	size mismatch for face_encoder.1.conv_block.1.bias: copying a param with shape torch.Size([64]) from checkpoint, the shape in current model is torch.Size([32]).
	size mismatch for face_encoder.1.conv_block.1.running_mean: copying a param with shape torch.Size([64]) from checkpoint, the shape in current model is torch.Size([32]).
	size mismatch for face_encoder.1.conv_block.1.running_var: copying a param with shape torch.Size([64]) from checkpoint, the shape in current model is torch.Size([32]).
	size mismatch for face_encoder.2.conv_block.0.weight: copying a param with shape torch.Size([64, 64, 3, 3]) from checkpoint, the shape in current model is torch.Size([32, 32, 3, 3]).
	size mismatch for face_encoder.2.conv_block.0.bias: copying a param with shape torch.Size([64]) from checkpoint, the shape in current model is torch.Size([32]).
	size mismatch for face_encoder.2.conv_block.1.weight: copying a param with shape torch.Size([64]) from checkpoint, the shape in current model is torch.Size([32]

**Comment by yihe1003** [comment]
As i can find any modification in the color_syncnet_train.py, can you share your sync_expert trainning code ? thank you

**Comment by primepake** [comment]
because we used 288x288, so we need to train sync_expert from scratch and then train wav2lip with sync_expert

**Comment by yihe1003** [comment]
> because we used 288x288, so we need to train sync_expert from scratch and then train wav2lip with sync_expert

ok thank you, as your suggestion, I retrained the sync_expert from scratch but the sync loss stays at 0.5 and can not drop. would you please kindly share your sync_expert weight that retraind on your 288X288 dataset? Thanks

**Comment by semchan** [comment]
> > because we used 288x288, so we need to train sync_expert from scratch and then train wav2lip with sync_expert
> 
> ok thank you, as your suggestion, I retrained the sync_expert from scratch but the sync loss stays at 0.5 and can not drop. would you please kindly share your sync_expert weight that retraind on your 288X288 dataset? Thanks

the same issue, did you fix it? thanks

**Comment by primepake** [comment]
the problem when syncnet doesn't converge is due to bad dataset, that mean you need to using syncnet_python to filter carefully with confidence around 6-9, and offset -3, 3
